src/analysis/object.rs

```diff
@@ -181,15 +161,11 @@ pub fn class(env: &Env, obj: &GObject, deps: &[library::TypeId]) -> Option<Info>
 
     // There's no point in generating a trait if there are no signals, methods, properties
     // and child properties: it would be empty
-    if generate_trait && !has_signals && !has_methods && properties.is_empty()
-        && child_properties.is_empty()
-    {
-        generate_trait = false;
-    }
+    //
+    // There's also no point in generating a trait for final types: there are no possible subtypes
+    let generate_trait = !final_type && (has_signals || has_methods || !properties.is_empty() || !child_properties.is_empty());
```

Comment:
`generate_trait` was already defined in beginning with big comment

Reply:
Yes, and here it's overridden now that we know whether signals/methods/properties are to be generated or not. Would you prefer if I call the first one `would_generate_trait` instead, to make the difference clearer that the second one is a definite one while the first just an intention (if we actually generate signals/methods/properties)?